feature-odyssey/src/main/kotlin/com/hedvig/android/odyssey/search/SearchViewModel.kt

```diff
@@ -0,0 +1,44 @@
+package com.hedvig.android.odyssey.search
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.hedvig.android.odyssey.model.ItemType
+import com.hedvig.android.odyssey.model.SearchableClaim
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+
+class SearchViewModel(
+  private val getClaimEntryPoints: GetClaimEntryPoints,
+) : ViewModel() {
+  private val searchableClaims = emptyList<SearchableClaim>()
+  private val _viewState = MutableStateFlow(SearchViewState(results = searchableClaims))
+  val viewState = _viewState
+
+  init {
+    viewModelScope.launch {
+      when (val result = getClaimEntryPoints.invoke()) {
+        is CommonClaimsResult.Error -> _viewState.update {
+          it.copy(
+            errorMessage = result.message,
+            isLoading = false,
+          )
+        }
+        is CommonClaimsResult.Success -> _viewState.update {
+          it.copy(
+            commonClaims = result.searchableClaims,
+            isLoading = false,
+          )
+        }
+      }
+    }
+  }
```

Comment:
This ATM means that at first load, if there's an error there's no way to reset this error state right? Always a bit tricky with VMs and doing stuff like this in `init{}`.
But I guess we also have no specs for a "retry" screen, so might have to keep it this way and have them go back and go into the screen again to handle such cases for now.

Reply:
Yeah no retry screen for now, we will hopefully get more elaborate designs soon!